src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/2.sql

```diff
@@ -364,6 +364,166 @@ CREATE NONCLUSTERED INDEX IX_ChangeFeed_StudyInstanceUid_SeriesInstanceUid_SopIn
+(
+    TagKey,
+    TagValue
+)
+
+/*************************************************************
+    Custom Tag Data Table for VR Types mapping to Double
+**************************************************************/
+CREATE TABLE dbo.CustomTagDouble (
+    TagKey                  BIGINT               NOT NULL, --PK
+    TagValue                FLOAT(53)            NOT NULL,
+    StudyKey                BIGINT               NOT NULL, --FK
+    SeriesKey               BIGINT               NULL,     --FK
+    InstanceKey             BIGINT               NULL,     --FK
+    Watermark               BIGINT               NOT NULL
+) WITH (DATA_COMPRESSION = PAGE)
+
+CREATE UNIQUE CLUSTERED INDEX IXC_CustomTagDouble ON dbo.CustomTagDouble
+(
+    TagKey,
+    StudyKey,
+    SeriesKey,
+    InstanceKey
+)
+
+CREATE NONCLUSTERED INDEX IXC_CustomTagDouble_TagKey_TagValue ON dbo.CustomTagDouble
+(
+    TagKey,
+    TagValue
+)
+
+/*************************************************************
+    Custom Tag Data Table for VR Types mapping to DateTime
+**************************************************************/
+CREATE TABLE dbo.CustomTagDateTime (
+    TagKey                  BIGINT               NOT NULL, --PK
+    TagValue                DATETIME2(7)         NOT NULL,
+    StudyKey                BIGINT               NOT NULL, --FK
+    SeriesKey               BIGINT               NULL,     --FK
+    InstanceKey             BIGINT               NULL,     --FK
+    Watermark               BIGINT               NOT NULL
+) WITH (DATA_COMPRESSION = PAGE)
+
+CREATE UNIQUE CLUSTERED INDEX IXC_CustomTagDateTime ON dbo.CustomTagDateTime
+(
+    TagKey,
+    StudyKey,
+    SeriesKey,
+    InstanceKey
+)
+
+CREATE NONCLUSTERED INDEX IXC_CustomTagDateTime_TagKey_TagValue ON dbo.CustomTagDateTime
+(
+    TagKey,
+    TagValue
+)
+
+/*************************************************************
+    Custom Tag Data Table for VR Types mapping to PersonName
+**************************************************************/
+CREATE TABLE dbo.CustomTagPersonName (
+    TagKey                  BIGINT               NOT NULL, --PK
+    TagValue                NVARCHAR(200)        COLLATE SQL_Latin1_General_CP1_CI_AI NOT NULL,
+    StudyKey                BIGINT               NOT NULL, --FK
+    SeriesKey               BIGINT               NULL,     --FK
+    InstanceKey             BIGINT               NULL,     --FK
+    Watermark               BIGINT               NOT NULL,
+    TagValueWords           AS REPLACE(REPLACE(TagValue, '^', ' '), '=', ' ') PERSISTED,
+) WITH (DATA_COMPRESSION = PAGE)
+
+CREATE UNIQUE CLUSTERED INDEX IXC_CustomTagPersonName ON dbo.CustomTagPersonName
+(
+    TagKey,
+    StudyKey,
+    SeriesKey,
+    InstanceKey
+)
+
+CREATE NONCLUSTERED INDEX IXC_CustomTagPersonName_TagKey_TagValue ON dbo.CustomTagPersonName
+(
+    TagKey,
+    TagValue
+)
+
+CREATE FULLTEXT INDEX ON CustomTagPersonName(TagValueWords LANGUAGE 1033)
```

Comment:
would this cause error we see last time?
Maybe we could postpone PersonName feature?